c#/Traverse/TraverseFolders/FileSystemVisitor.cs

```diff
@@ -1,72 +1,71 @@
-public delegate bool ReturnedDelegate(string filename, string extention);
-
+public delegate bool FilterFunctionDelegate(string name);
+public delegate void Notify();
+public delegate void FindFolderDelegate(string filename,CustomEventArgs eventArgs);
 class FileSystemVisitor
 {
-    public FileSystemVisitor(string filename)
+    private readonly string _path;
+    private readonly FilterFunctionDelegate _filterFunc;
+
+    public event Notify OnStart;
+    public event Notify OnFinish;
+    public event FindFolderDelegate OnFileFound;
+    public event FindFolderDelegate OnFolderExclude;
```

Comment:
Add 2 more events for DirectoryFound and FilteredFileFound and subscribe handlers to them, just to satisfy the task condition. And name all the events as they described in the task.